### PR TITLE
#244 - work with Authentication.User instead of username

### LINKED ISSUE #244: Change auth interfaces signature
Change signature of `Identities`, `Permission` and `Permissions` interfaces to work with `Authentication.User` class interad of just user name.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #244 
Made `Identities`, `Permission` and `Permissions` work with `Authentication.User` instead of user name.

### CODE REVIEW

File: src/main/java/com/artipie/http/auth/Authentication.java [hunk 1]
```diff
@@ -38,6 +38,11 @@
  */
 public interface Authentication {
 
+    /**
+     * Super user instance.
+     */
+    User SUPER_USER = new User("*");
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/auth/Authentication.java:44]: @olenagerasimova meaning of "super" user is unclear. It seams that it's purpose is something like "any user" and is used specifically to check permissions for any user in `Permissions`. So I'd suggest to change the name and move the constant to `Permissions`

File: src/main/java/com/artipie/http/auth/Permissions.java [hunk 2]
```diff
@@ -37,11 +37,11 @@
 
     /**
      * Check if user allowed to perform action.
-     * @param name User name
+     * @param user User name
```
Comment (olegmoz) [inline src/main/java/com/artipie/http/auth/Permissions.java:40]: @olenagerasimova comment for the argument needs an update too

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see two comments

**Comment by olenagerasimova** [comment]
@olegmoz thanks, all fixed

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/http#245` is not assigned, can't get performer

**Pull request merged**